### PR TITLE
Adding auth_header option to inject a jwt bearer authentication header

### PR DESCRIPTION
**Pull request by jiggo**

### CODE REVIEW

**Comment by Serhioromano** [comment]
I get it. this is nice idea. Unfortunately code you PR is not very practical or so to say flexible. It only use Bearer type. But there are others. Would it be nice to be able to extend not only headers but anything?

I think we have to add `beforeSend ` callback. And there any user will be able to add any header or even parameters.

What do you think? 

Could change `auth_header` option to `before_events_request` and make it a callback? Also add something to readme file so people could know about that.

**Comment by jiggo** [comment]
You're right, I think it's a better and more flexible way to implement a callback in order to define any possible header. I'm quite new to the contribution mechanism of github so l have produced a code that fits to my particular situation, absolutely not flexible indeed.

I'll modify the code accordingly to your suggestions as soon as possible.

**Comment by Serhioromano** [comment]
All you need to do is to apply changes to master and push it to your fork on gitHub. those changes will appear here automatically. You do not need to do another PR.

**Comment by jiggo** [comment]
Ok!

**Comment by eleazan** [comment]
Nice!

File: js/calendar.js [hunk 1]
```diff
@@ -877,6 +878,10 @@ if(!String.prototype.formatNum) {
 	Calendar.prototype._loadEvents = function() {
 		var self = this;
 		var source = null;
+		var auth_header = null;
+		if('auth_header' in this.options && this.options.auth_header !== '') {
+			auth_header = this.options.auth_header;
+		}
```
Comment (Serhioromano) [inline js/calendar.js]: I do not think you need this 881-884 block of code. You have already set it in defaults.

File: js/calendar.js [hunk 2]
```diff
@@ -911,7 +916,11 @@ if(!String.prototype.formatNum) {
 							url: buildEventsUrl(source, params),
 							dataType: 'json',
 							type: 'GET',
-							async: false
+							async: false,
+							beforeSend: function(request){
+								if(auth_header.length)
```
Comment (Serhioromano) [inline js/calendar.js]: Simply use `self.this.options.auth_header` instead.

**Pull request closed without merging**